src/main/scala/edu/ie3/powerFactory2psdm/converter/types/TransformerType2WConverter.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.powerFactory2psdm.converter.types
+
+import edu.ie3.datamodel.models.input.connector.`type`.Transformer2WTypeInput
+import edu.ie3.powerFactory2psdm.exception.pf.{
+  ConversionException,
+  ElementConfigurationException
+}
+import edu.ie3.powerFactory2psdm.model.powerfactory.types.TransformerType2W
+import tech.units.indriya.quantity.Quantities
+import tech.units.indriya.unit.Units.{OHM, PERCENT, SIEMENS, VOLT}
+import edu.ie3.util.quantities.PowerSystemUnits.{DEGREE_GEOM, VOLTAMPERE}
+
+import math.{pow, sqrt}
+import java.util.UUID
+import javax.measure.MetricPrefix
+
+object TransformerType2WConverter {
+
+  def convert(input: TransformerType2W): Transformer2WTypeInput = {
+
+    val sRated = input.sRated * 1e6
+    val vRatedA = input.vRatedA * 1e3
+    val vRatedB = input.vRatedB * 1e3
+    val pCu = input.pCu * 1e3
+    val pFe = input.pFe * 1e3
+    val uk = (input.uk / 100) * vRatedA
+    val iRated = sRated / (math.sqrt(3) * vRatedA)
+    val iNoLoadNom = (input.iNoLoad / 100) * iRated
+
+    // short circuit experiment
+    val rSc = pCu / (3 * pow(iRated, 2))
+    val zSc = (uk / sqrt(3)) / iRated
+    if (rSc > zSc) {
+      throw ConversionException(
+        s"Short circuit experiment calculations of 2w transformer type: ${input.id} is not possible due to faulty " +
+          s"parameters. The short circuit resistance can't exceed the short circuit impedance."
+      )
+    }
+    val xSc = sqrt(pow(zSc, 2) - pow(rSc, 2))
+
+    // no load experiment
+    val yNoLoad = iNoLoadNom / (vRatedA / sqrt(3))
+    val gNoLoad = pFe / pow(vRatedA, 2)
+    val bNoLoad = sqrt((pow(yNoLoad, 2) - pow(gNoLoad, 2)).doubleValue)
+    if (gNoLoad > yNoLoad) {
+      throw ConversionException(
+        s"No load experiment calculations of 2w transformer type: ${input.id} is not possible due to faulty parameters." +
+          s"The no load conductance can't exceed the no load admittance."
+      )
+    }
```

Comment:
Has to be moved one line up, otherwise it does not prevent a mathematical error in the root calculation above.

Reply:
.. makes sense 😄